app/frontend/basic-auth.js

```diff
@@ -21,25 +15,22 @@ const basicAuthChallenge = (usernamePlain, passwordPlain) => {
     });
   }
 
-  const validUsername = basicAuth.safeCompare(usernamePlain, authUsername);
+  const validUsername = basicAuth.safeCompare(plainUsername, authUsername);
 
   let validPassword = false;
   if (auth2FAEnabled && SECRET_2FA) {
-    const validPasswordPlain = basicAuth.safeCompare(
-      passwordPlain.slice(0, -6),
-      config.frontend.auth.password
+    const validPlainPassword = basicAuth.safeCompare(
+      plainPassword.slice(0, -6), // Remove last 6 digits for OTP code
+      authPassword
     );
 
     const validPassword2FA = Boolean(
-      otp.verifyToken(SECRET_2FA, passwordPlain.replace(authPassword, ''))
+      otp.verifyToken(SECRET_2FA, plainPassword.replace(authPassword, ''))
     );
 
-    validPassword = validPasswordPlain && validPassword2FA;
+    validPassword = validPlainPassword && validPassword2FA;
```

Comment:
👍

Reply:
Same thing, please check the updated code.